src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -307,7 +307,13 @@ static void uct_rdmacm_cm_handle_error_event(struct rdma_cm_event *event)
     ucs_status_t status;
 
     if (event->event == RDMA_CM_EVENT_REJECTED) {
-        status    = UCS_ERR_REJECTED;
+        if (cep->flags & UCT_RDMACM_CM_EP_ON_SERVER) {
+            status = UCS_ERR_CONNECTION_RESET;
```

Comment:
how can we have rejected event on server side?

Reply:
e.g pls document it in the code

Reply:
RDMA_CM_EVENT_REJECTED
Indicates that a **connection request** or **response** was rejected by the remote end point. The event status field will contain the transport specific reject reason if available. Under InfiniBand, this is the reject reason carried in the IB CM REJ message.

Reply:
> e.g pls document it in the code

OK

Reply:
Hmm.. maybe when @alinask adds uct_cm_establish(ucs_status) API we would still need REJECTED on server side from UCT.. WDYT?

Reply:
1) rdmacm can generate reject event without user involvement as well.
2) then we'll need to pass some private data to indicate the reason of reject event.
3) IMO `Connection reset by remote peer` is clearer for server if client breaks connection establishment the middle of it. On other hand rdmacm states that response can be "rejected" which could match `uct_reject` or `uct_disconnect`. Last one looks clearer for me, we are able to do correct translation uct_disconnect -> rdmacm_reject by internal EP state.

Reply:
ucs_status_str(UCS_ERR_CONNECTION_RESET) == "Connection reset by remote peer"

Reply:
ok, i guess that it will be in line with TCP this way (using CONNECTION_RESET)

Reply:
> e.g pls document it in the code

done